### DISCUSSION TITLE
Flashing complete but not connecting afterwards.

### DISCUSSION THREAD

**Discussion by yousefmansy1** [emporia-vue-local/esphome#3] [Q&A] [answered]
I was able to follow all the instructions but after flashing the bootloader it does not show up in my esphome dashboard or even connected to my router at all.

- I have confirmed the SSID and password for my wifi ap in the config file are correct
- I was able to do the flashing through ESPHome Web and observed the same behavior
- If I run "prepare for adoption" from ESPHome web it'll say it finished successfully only to throw a `An error occurred. Improv Wi-Fi Serial not detected` error. Then, no devices show up on my ESPHome dashboard and no esp device on my network

I'm new to ESPHome and am at a loss on what do to.

```ps
PS C:\Users\Yousef\Desktop\Newfolder> python -m esphome run vue2.yaml 
INFO Reading configuration vue2.yaml...
INFO Detected timezone 'America/Los_Angeles'
INFO Generating C++ source...
INFO Compiling app...
Processing emporiavue2 (board: esp32dev; framework: espidf; platform: platformio/espressif32 @ 3.5.0)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 (3.5.0) > Espressif ESP32 Dev Module
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-espidf 3.40302.0 (4.3.2)
 - tool-cmake 3.16.4
 - tool-esptoolpy 1.30100.210531 (3.1.0)
 - tool-idf 1.0.1
 - tool-mconf 1.4060000.20190628 (406.0.0)
 - tool-ninja 1.9.0
 - toolchain-esp32ulp 1.22851.191205 (2.28.51)
 - toolchain-riscv32-esp 8.4.0+2021r2-patch2
 - toolchain-xtensa-esp32 8.4.0+2021r2-patch2
 - toolchain-xtensa-esp32s2 8.4.0+2021r2-patch2
Reading CMake configuration...
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ off, Compatibility ~ soft
Found 0 compatible libraries
Scanning dependencies...
No dependencies
Building in release mode
Retrieving maximum program size .pioenvs\emporiavue2\firmware.elf
Checking size .pioenvs\emporiavue2\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [=         ]  10.5% (used 34256 bytes from 327680 bytes)
Flash: [====      ]  40.7% (used 747137 bytes from 1835008 bytes)
========================================================================================================= [SUCCESS] Took 18.93 seconds =========================================================================================================INFO Successfully compiled program.
Found multiple options, please choose one:
  [1] COM4 (Silicon Labs CP210x USB to UART Bridge (COM4))
  [2] Over The Air (emporiavue2.local)
(number): 1
esptool.py v3.3.1
Serial port COM4
Connecting.....
Chip is ESP32-D0WD (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 98:cd:ac:b1:18:4c
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Auto-detected Flash size: 8MB
Flash will be erased from 0x00010000 to 0x000c6fff...
Flash will be erased from 0x00001000 to 0x00007fff...
Flash will be erased from 0x00008000 to 0x00008fff...
Flash will be erased from 0x0000d000 to 0x0000efff...
Compressed 747520 bytes to 487004...
Wrote 747520 bytes (487004 compressed) at 0x00010000 in 11.6 seconds (effective 515.8 kbit/s)...
Hash of data verified.
Flash params set to 0x0230
Compressed 26288 bytes to 16033...
Wrote 26288 bytes (16033 compressed) at 0x00001000 in 0.8 seconds (effective 250.1 kbit/s)...
 0.1 seconds (effective 288.0 kbit/s)...
Hash of data verified.
Compressed 8192 bytes to 31...
Wrote 8192 bytes (31 compressed) at 0x0000d000 in 0.0 seconds (effective 1638.6 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
INFO Successfully uploaded program.
INFO Starting log output from COM4 with baud rate 115200

```

**Comment by yousefmansy1** [comment, marked as answer]
Ok, nevermind
I did a lot of troubleshooting beforehand and thought "Hey why not just install it with the AC power?"
Then it showed up in my router 🙃 
It looks like it only connects with the main power and not with the power from serial adapter.

Still though I ran into an issue with resolving the name `emporiavue2.local` to the actual IP
```
WARNING Can't connect to ESPHome API for emporiavue2.local: Error resolving IP address
```

So I assigned it a static IP in my router and defined an address with dnsmasq,
Then it finally worked.

Leaving this up in case someone else runs into the same issue.

**Reply by flaviut** [reply]
> Hey why not just install it with the AC power?

Yup! The wifi radio draws a lot of power. I've found that most USB↔serial converters are unable to supply enough power for that.

> Still though I ran into an issue with resolving the name emporiavue2.local to the actual IP

Never ran into this before, but debugging networks is not fun. I think you found the most reasonable solution.

**Reply by Jens5395** [reply]
I had the same issue - flashed it without AC connected, reset it (still powered by the serial converter) and the Emporia did not come on line. I determined that the Emporia was continually resetting due to brown-outs. Once AC power was set up, everything worked just fine.

**Reply by Finagain** [reply]
I did everything suggested (Static IP, connected to mains) and it still isn't showing up neither on the router or in ESPHome.  I thought it might be a bad unit but for fun, I re-flashed with the original firmware and it popped up on the network.  I can't get ESPHome to work on this thing.  I even tried updating my ESPhome to 2022.8.2 and then down to 2022.1.2 (the same version the guy uses on YouTube) and nothing works.  I'm not new to ESPhome and have several devices working currently.  What version of ESPhome are you guys working with?

**Reply by flaviut** [reply]
Hey @Finagain, it sounds like you're having a different issue. Would you please create a new thread with exactly what you've written here?

**Reply by Finagain** [reply]
> Hey @Finagain, it sounds like you're having a different issue. Would you please create a new thread with exactly what you've written here?

Done.

**Comment by mwaterbu** [comment]
I know this is a fairly old question, but I would like to suggest an alternative solution to connect when still on serial (the current accepted answer of connecting the AC wiring harness is valid as well, but being able to test with just the USB serial is more convenient before the install is done).
When I finished flashing via a USB FT232 adapter, the device did not come online at all. It did not connect to WiFi, and I could not even real the logs over the serial connection. Power cycling with the serial adapter did not help.
Eventually I remembered that holding GPIO0 to ground when powering up puts the ESP in boot mode, so it was just waiting for a new flash, instead of running the code I just flashed on. I unplugged my jumper on IO0, power cycled again, and it worked immediately. Hope this can help someone else in the same situation.
